tests/compile-fail/uninit_raw_ptr.rs

```diff
@@ -0,0 +1,4 @@
+fn main() {
+    let _val = unsafe { std::mem::MaybeUninit::<*const str>::uninit().assume_init() };
```

Comment:
```suggestion
    let _val = unsafe { std::mem::MaybeUninit::<*const u8>::uninit().assume_init() };
```
`str` is an unsized type, making it rather special, so let's use a more standard type here.